### PR TITLE
Added support of EG ECS Roll and Import stateful Instance API's

### PR DESCRIPTION
**Pull request by janetlinkiruba**
Checklist
---------

JIRA Ticket -

Checklist (Please mark Yes/No/NA for each of the questions below) -

- I have provided clear and detailed explanation of changes made in the proposed PR : Yes
- I have compiled the code successfully my laptop : Yes
- I have written examples to demonstrates the usage of the newly implemented APIs : Yes
- I have incremented the release version : No
- I have run the Intellij Code Analysis on newly added code and fixed all (valid) reported issues : No

OR

- I have taken approval from scrum master or project manager for skipping any of the items above : No

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 1]
```diff
@@ -180,6 +180,29 @@ public static void main(String[] args) throws IOException {
...
+        System.out.println("----------List cluster Rolls--------------");
+        List<ElastigroupEcsClusterRollResponse> listClusterRolls = getAllClusterRolls(elastigroupClient, "group-id");
+
+        //Update cluster Roll
+        System.out.println("----------Update cluster Roll--------------");
+        ElastigroupEcsClusterRollResponse updateRollResponse = updateClusterRoll(elastigroupClient, "group-id", "STOPPED");
+
+        //Create Instance Signal
+        System.out.println("----------Create Instance Signal--------------");
+        Boolean createInstanceSignalStatus = createInstanceSignal(elastigroupClient);
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:201]: Should mention what signal is to be created.

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 2]
```diff
@@ -1132,4 +1155,74 @@ private static List<GetInstanceTypesResponse> getSuggestedInstanceTypes(Spotinst
...
+
+        System.out.println(String.format("Update Cluster Roll. ClusterId: %s", groupId));
+        ElastigroupEcsClusterRollResponse response = client.updateECSClusterRollinEG(updateRoll, groupId);
+
+        System.out.println(String.format("RollStatus: %s", response.getStatus()));
+
+        return response;
+    }
+
+    private static Boolean createInstanceSignal(SpotinstElastigroupClient elastigroupClient) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:1206]: Type of signal to be passed as parameter.

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java [hunk 3]
```diff
@@ -836,4 +836,204 @@ public RepoGenericResponse<List<GetInstanceTypesResponse>> getSuggestedInstanceT
 
         return retVal;
     }
+
+    @Override
+    public RepoGenericResponse<ElastigroupEcsClusterRollResponse> initiateClusterRollInEGWithECS(ElastigroupEcsInitiateRoll rollRequest, String groupId, String authToken, String account) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:841]: Can we move this repo class to repo folder.

File: src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/aws/ElastigroupTerminateStatefulInstanceImportResponse.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ElastigroupTerminateStatefulInstanceImportResponse {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/aws/ElastigroupTerminateStatefulInstanceImportResponse.java:12]: should be in response folder

**Review by anuragsharma-123 — CHANGES_REQUESTED** [review]
Please review comments and rework.

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java (hunk 1 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:201]: fixed

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java (hunk 2 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:1206]: fixed

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java (hunk 3 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupRepo.java:841]: showing so many conflicts during refactor, will do it later

File: src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/aws/ElastigroupTerminateStatefulInstanceImportResponse.java (hunk 4 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/aws/ElastigroupTerminateStatefulInstanceImportResponse.java:12]: This is a response object class

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**